internal/engine/configs/configs_controller.go

```diff
@@ -101,9 +100,11 @@ func (cc *ConfigsController) needsBuild(ctx context.Context, st store.RStore) (b
 		return buildEntry{}, false
 	}
 
-	filesChanged := make([]string, 0, len(state.PendingConfigFileChanges))
-	for k := range state.PendingConfigFileChanges {
-		filesChanged = append(filesChanged, k)
+	filesChanged := []string{}
+	for _, st := range state.TiltfileState.BuildStatuses {
+		for k := range st.PendingFileChanges {
+			filesChanged = append(filesChanged, k)
```

Comment:
Is it possible for this to result in dupes now? (I don't think that'll actually harm anything even if it does)

Reply:
nope! i don't think so! but the data model supports multiple Target Statuses, and certainly could imagine a future change where it does matter.